### ISSUE TITLE
[Bug]: CRA connection stays off and CRA company stays unknown with active CRA in Ajax app

### ISSUE THREAD

**Issue by bogar** [bvis/aegis-hass#154]
### Describe the bug
On a live installation with an active CRA company shown in the Ajax mobile app, the integration reports no CRA data:

- `CRA connection` stays `off`
- `CRA company` stays `unknown`
- `approved_companies`, `pending_approval_companies`, and `pending_removal_companies` are all empty

This happens on the current upstream release (`v1.4.5`).

The installation previously exposed CRA data correctly. After updating to newer 1.4.x versions, the CRA-related entities no longer reflect what the Ajax app shows.

### Steps to reproduce
1. Install `Aegis for Ajax` via HACS.
2. Configure the integration with app label `Ajax`.
3. Use an account / space where the Ajax mobile app shows an active CRA company.
4. Enable the CRA diagnostic sensor.
5. Reload the integration or restart Home Assistant.
6. Check the CRA entities in Home Assistant.

### Expected behavior
If the Ajax app shows an active CRA company, Home Assistant should expose corresponding CRA state:

- `CRA connection` should not remain `off`
- `CRA company` should not remain `unknown`
- company lists should not remain empty

### Home Assistant version
2026.5.2

### Integration version
1.4.5

### Installation method
HACS

### App label (co-branded app)
Ajax

### FCM push notifications configured?
Yes

### Relevant logs
Debug logging was enabled for `custom_components.aegis_ajax` before reloading the integration.

The integration starts successfully, restores the session, starts the device stream, connects HTS, requests full hub settings/status, and starts FCM. However, there is still no CRA data exposed in entities after startup.

Sanitized log excerpt:

```text
2026-05-18 15:47:13.595 DEBUG [custom_components.aegis_ajax] Restoring stored Ajax session for entry ENTRY_ID (user=USER_ID)
2026-05-18 15:47:13.970 DEBUG [custom_components.aegis_ajax.coordinator] Device stream started for space SPACE_ID
2026-05-18 15:47:13.970 INFO  [custom_components.aegis_ajax.coordinator] Aegis startup: device streams 1/1 started, HTS lifecycle scheduled
2026-05-18 15:47:13.971 DEBUG [custom_components.aegis_ajax.coordinator] Finished fetching aegis_ajax data in 0.355 seconds (success: True)
2026-05-18 15:47:14.241 DEBUG [custom_components.aegis_ajax.notification] FCM token obtained, registering with Ajax servers
2026-05-18 15:47:14.300 DEBUG [custom_components.aegis_ajax.notification] Push token registered with Ajax servers
2026-05-18 15:47:14.300 INFO  [custom_components.aegis_ajax.notification] FCM push client started — push notifications active
2026-05-18 15:47:14.501 DEBUG [custom_components.aegis_ajax.api.hts.client] HTS authenticated: 1 hub(s), token=REDACTED...
2026-05-18 15:47:14.501 INFO  [custom_components.aegis_ajax.coordinator] HTS connected, 1 hub(s)
2026-05-18 15:47:14.604 DEBUG [custom_components.aegis_ajax.api.hts.client] Sent REQUEST_FULL_SETTINGS to HUB_ID
2026-05-18 15:47:14.606 DEBUG [custom_components.aegis_ajax.api.hts.client] Sent REQUEST_FULL_STATUS to HUB_ID
2026-05-18 15:47:14.689 DEBUG [custom_components.aegis_ajax.api.hts.client] Hub HUB_ID: parsed 137 keys from SETTINGS_BODY
2026-05-18 15:47:14.737 DEBUG [custom_components.aegis_ajax.api.hts.client] Hub HUB_ID: parsed 55 keys from STATUS_BODY
```

Current entity state after reload:

```text
CRA connection = off
CRA company = unknown
approved_companies = []
pending_approval_companies = []
pending_removal_companies = []
```

Notably, the debug output does not show any CRA / monitoring-specific values (`monitoring_active`, `cms_active`, or `monitoring_companies`), so it is hard to tell whether the account currently receives no CRA payloads or whether the integration is dropping them.

### Screenshots
Not included.

### Additional context
- The Ajax mobile app currently shows an active CRA company for this installation.
- Home Assistant Core is up to date on `2026.5.2`.
- The integration is installed from HACS and reports `v1.4.5`.
- `check_config` is valid.
- No duplicate `custom_components/aegis_ajax` copies are present.
- Hub model: `Hub Two Plus`.
- In the same installation, other Ajax entities are healthy (HTS connectivity sensors and firmware update entity look normal).

**Comment by bvis** [comment]
Thanks for the precise report and the sanitized log. Two paths feed those CRA entities and your log shows both failing silently, so let's narrow which one before assuming a parser regression:

1. **Hub-level signal** (`monitoring.cms_active`) — comes from the HTS `STATUS_BODY` we see your install parsed (`55 keys` line). The binary sensor reads it as `hub.statuses["monitoring_active"]`. The fastest way to tell whether this key is present in your 55 parsed keys is **Settings → Devices & Services → Aegis for Ajax → ⋮ → Download diagnostics**. In the resulting JSON, under `devices.<your-hub-id>.statuses`, look for `"monitoring_active"`. Could you attach (or grep+share) just that block? If the key is missing, the firmware on `Hub Two Plus` isn't emitting the `monitoring` status oneof in this snapshot and the fallback path takes over.

2. **Space-level signal** (`monitoring_companies`) — populated from the `SpaceService` snapshot we call on rooms refresh. The diagnostics dump in `v1.4.5` doesn't expose this directly (gap on my side — I'll widen it), but two questions help triangulate:
   - In the Ajax mobile app, what is the CRA company's exact status — **Approved**, **Pending approval**, or **Pending removal**? Only `Approved` flips `has_monitoring` to true; the other two only show up under `pending_*` attributes.
   - Are you using the **same Ajax account** in HA as the one that sees the CRA company in the app? CRA visibility is per-user; an invited user with limited permissions may not receive it in the `SpaceService` snapshot.

3. **Which version was the last one that exposed CRA correctly?** (and approximately when you upgraded away from it). Between `1.2.3-beta.1` and `1.4.3` the binary sensor was sourcing CRA from `has_monitoring` only — `1.4.4` restored `monitoring.cms_active` as the primary signal with `has_monitoring` as fallback. Knowing the last-good version tells me whether you're hitting the same regression we thought was closed or a different one.

With those three pieces (diagnostics excerpt, app-side CRA status, last-good version) I can tell whether this is a parser gap, a per-account visibility issue, or a real regression in the `1.4.4` fix that didn't actually cover your case.

**Comment by bogar** [comment]
I pulled the diagnostics JSON and checked the hub block under `data.devices.<hub-id>.statuses`.

The important part is that `monitoring_active` is **not present** there.

Sanitized excerpt:

```json
"<hub-id>": {
  "name": "<redacted>",
  "type": "hub_two_plus",
  "state": "online",
  "online": true,
  "malfunctions": 0,
  "bypassed": false,
  "battery": {
    "level": 100,
    "low": false
  },
  "statuses": [
    "mobile_network_type",
    "gsm_connected"
  ]
}
```

So on this installation, the hub-level path currently exposes only `mobile_network_type` and `gsm_connected` in diagnostics, and does **not** expose `monitoring_active`.

A few other relevant points:

- In the Ajax mobile app, the CRA/security company is shown as **active**.
- Home Assistant is using a **different Ajax account** than the one I use in the mobile app.
- However, with this same HA account, CRA data **was visible in older integration versions**.
- The diagnostics JSON shows a single configured space and otherwise healthy integration state; there is nothing obvious there suggesting a broken setup.

That makes it look less like a generic permission problem and more like either:
1. the current HTS snapshot for this account / hub no longer exposes the hub-level monitoring flag, and
2. the space-level fallback is also not surfacing CRA data for this HA account anymore.

If it helps, I can next provide a sanitized diagnostics attachment or test a branch with extra debug around `monitoring_active` / `monitoring_companies`.

**Comment by bogar** [comment]
I reverted back to version 1.2.3-beta.2 and both work on this version.

**Comment by bvis** [comment]
Hi @bogar — quick update. The CRA-company regression you were hitting was caused by the `CLIENT_VERSION` header we send to Ajax: bumping it to `3.46` in a later release silently made the backend drop `monitoring_companies` from the snapshot. We've pinned it back to `3.30` in `1.5.0-beta.1` (and the just-released `1.5.0-beta.2`).

Would you mind enabling "Show beta versions" in HACS, updating to `1.5.0-beta.2`, restarting HA, and checking whether your CRA company name + connection sensors come back? That should match what you saw in `1.2.3-beta.2`, and if it does we can promote the fix to stable. If anything still looks off, the same `Download diagnostics` JSON would help.

Thanks for the patience.

**Comment by bogar** [comment]
Tested here with 1.5.0-beta.3 and the CRA entities are working correctly again. Both the CRA connection sensor and the CRA company sensor are back to the expected values.

Thanks for tracking this down and for the quick fix. Closing the issue since it is resolved on my installation.
